### PR TITLE
input: use K_KERNEL_STACK_SIZEOF to set the thread stack size

### LINKED ISSUE #66821: [Coverity CID: 338057] Out-of-bounds access in drivers/input/linux_evdev.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/master/drivers/input/linux_evdev.c

Category: Memory - corruptions
Function: `linux_evdev_init`
Component: Drivers
CID: [338057](https://scan9.scan.coverity.com/reports.htm#v29726/p12996/mergedDefectId=338057)

Details:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/input/linux_evdev.c#L100

Please fix or provide comments in coverity using the link:

https://scan9.scan.coverity.com/reports.htm#v29271/p12996.

For more information about the violation, check the [Coverity Reference](https://scan9.scan.coverity.com/doc/en/cov_checker_ref.html#static_checker_OVERRUN). ([CWE-119](http://cwe.mitre.org/data/definitions/119.html))

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the MAINTAINERS file.

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Use K_KERNEL_STACK_SIZEOF instead of the config directly to set the stack size in k_thread_create() calls.

---
Not totally sure this is the actual thing coverity is complaining about but the issue is not very clear, this is the only thing I can think of, that's how other drivers do it and it's certainly not a bad idea.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Yeah nevermind I did this based on `can_native_linux.c` but now I see https://github.com/zephyrproject-rtos/zephyr/issues/66822 :-)

Well, probably worth doing this change regardless.

**Comment by thedjnK** [comment]
I think there's something wrong with the macros in certain conditions, see https://github.com/zephyrproject-rtos/zephyr/issues/65584 ping @andyross

**Review by aescolar — APPROVED** [review]
Ok. This change does no harm, and it would seem to be better (if K_KERNEL_STACK_RESERVED!=0).

But this coverity report seems quite bogus (as it seems to think K_KERNEL_STACK_RESERVED = 592 for  [drivers/input/linux_evdev.c](https://github.com/zephyrproject-rtos/zephyr/pull/66857/files#diff-54226aad7a01c512fd593a1c7b66a8397803d4b81edd404623da7d1db3376c9a) while there it can only be 0. So I'd guess it is mixing that value between archs(?)

**Comment by fabiobaltieri** [comment]
@aescolar yeah, dropped the "fixed" tags, the coverity warning fired for quite few other cases, we'll have to follow up separately.

**Comment by andyross** [comment]
Sorry, was out skiing earlier this week[1].  The coverity report in https://github.com/zephyrproject-rtos/zephyr/issues/65584 and (presumably) here looked legitimate: CONFIG_KERNEL_STACK_RESERVED can be set to values that are larger than the actual stack being statically allocated (some of our test stacks are extremely small, which is actually a good thing because it teases out issues like this).  And we either don't have validity checking for this or we have it at a spot coverity can't see through.

Now, maybe the actual kconfig isn't ever set to the value coverity thinks it is, so there may be circumstances where it's throwing a false positive in test code that can't overflow as written.  But that kconfig **can** have any value and be emitted into any downstream application, and we want not to crash.

Basically this smells like a kernel validation issue in thread creation somewhere to me.  I think we want to find it and not just squash the warnings.

[1] Came back a day early though.  Terrible snow season in Idaho so half the mountain was closed.

**Review by andyross — APPROVED** [review]
But no reason not to merge this.  K_KERNEL_STACK_SIZEOF is indeed the correct API for this, though most apps can get away with passing the raw number.

**Comment by fabiobaltieri** [comment]
Thanks @andyross, there's a few of these and the information is getting spread around, I'll merge this then make an umbrella issue next week so can track them.

**Comment by henrikbrixandersen** [comment]
> Thanks @andyross, there's a few of these and the information is getting spread around, I'll merge this then make an umbrella issue next week so can track them.

Umbrella issue: https://github.com/zephyrproject-rtos/zephyr/issues/67106
